org.eclipse.scanning.connector.epicsv3/src/org/eclipse/scanning/connector/epicsv3/EpicsV3DynamicDatasetConnector.java

```diff
@@ -518,39 +505,62 @@ public void monitorChanged(MonitorEvent arg0) {
 							}
 							lastSystemTime = timeNow;
 						}
-						
-					} else if (channelName.equalsIgnoreCase(currentWidthPV)) {
-						DBR dbr = arg0.getDBR();
+					} else if (channelName.equalsIgnoreCase(dim0PV)) {
 						DBR_Int dbri = (DBR_Int)dbr;
-						if (width != dbri.getIntValue()[0]) {
-							width = dbri.getIntValue()[0];
-							dataChannelMonitor.removeMonitorListener(dataChannelMonitorListener);
-							int dataSize = height * width;
-							if (numDimensions == 3) {
-								dataSize *= rgbChannels;
-							}
-							dataChannelMonitor = ec.setMonitor(dataChannel, dataChannelMonitorListener, dataSize);
+						int value = dbri.getIntValue()[0];
+						if (dim0 != value) {
+							dim0 = value;
+							logger.debug("New value {} for {}", value, dim0PV);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "New value {} for {}" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)